Content.Server/GameObjects/Components/Janitorial/LightReplacerComponent.cs

```diff
@@ -0,0 +1,188 @@
+#nullable enable
+using Content.Server.GameObjects.Components.Items.Storage;
+using Content.Server.GameObjects.Components.Power.ApcNetComponents.PowerReceiverUsers;
+using Content.Shared.GameObjects.EntitySystems.ActionBlocker;
+using Content.Shared.Interfaces;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.Containers;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Localization;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom;
+using Robust.Shared.ViewVariables;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Content.Server.GameObjects.Components.Janitorial
+{
+    /// <summary>
+    ///     Device that allows user to quikly change bulbs in <see cref="PoweredLightComponent"/>
+    ///     Can be reloaded by new light tubes or light bulbs
+    /// </summary>
+    [RegisterComponent]
+    public class LightReplacerComponent : Component, IAfterInteract, IInteractUsing
+    {
+        public override string Name => "LightReplacer";
+
+        [DataField("contents")] private List<LightReplacerEntity> _contents = new();
+        [DataField("sound")] private string _sound = "/Audio/Weapons/click.ogg";
+
+        [ViewVariables] private IContainer _bulbsStorage = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            // fill light replacer container with bulbs
+            _bulbsStorage = ContainerHelpers.EnsureContainer<Container>(Owner, "light_replacer_storage");
+            foreach (var ent in _contents)
+            {
+                for (var i = 0; i < ent.Amount; i++)
+                {
+                    var bulb = Owner.EntityManager.SpawnEntity(ent.PrototypeName, Owner.Transform.Coordinates);
+                    _bulbsStorage.Insert(bulb);
+                }
+            }
```

Comment:
You should clear the list after this

Reply:
Do you mean _contents.Clear()?

Reply:
Do you mean _contents.Clear()?

Reply:
Yes

Reply:
I would recommend deferring spawning these like guns do if possible given you're spawning 14 entities in 1.

Reply:
I think that this can probably be split off into an EnsureContainerFillComponent or something like that.

Reply:
Yeah most likely, we'll definitely want the optimisation prior to release..